src/equations/compressible_euler_multicomponent_2d.jl

```diff
@@ -7,61 +7,58 @@
 
 Multicomponent version of the compressible Euler equations for an ideal gas in two space dimensions.
 """
-struct CompressibleEulerMulticomponentEquations2D{RealT<:Real} <: AbstractCompressibleEulerMulticomponentEquations{2, 5}
-  gamma1          ::RealT 
-  gamma2          ::RealT
-  gas_constant1   ::RealT
-  gas_constant2   ::RealT
-  cv1             ::RealT
-  cv2             ::RealT
-  cp1             ::RealT
-  cp2             ::RealT
-end
 
-function CompressibleEulerMulticomponentEquations2D()
+struct CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS} <: AbstractCompressibleEulerMulticomponentEquations{2, NVARS}
+  gamma             ::SVector{NCOMP, Real} 
+  gas_constant      ::SVector{NCOMP, Real}
+  cv                ::SVector{NCOMP, Real}
+  cp                ::SVector{NCOMP, Real}
```

Comment:
That's probably terrible for performance since `Real` is an abstract type. You should rather use something like
```suggestion
struct CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS, RealT<:Real} <: AbstractCompressibleEulerMulticomponentEquations{2, NVARS}
  gamma             ::SVector{NCOMP, RealT} 
  gas_constant      ::SVector{NCOMP, RealT}
  cv                ::SVector{NCOMP, RealT}
  cp                ::SVector{NCOMP, RealT}
```
and adjust the inner constructor etc. Then, you can use types with `RealT === Float64` etc. to get good performance.

Reply:
Done in c0314d0.